plugins/modules/ipaservice.py

```diff
@@ -462,8 +618,16 @@ def main():
     state = ansible_module.params_get("state")
 
     # check parameters
+    if (names is None or len(names) < 1) and \
+       (services is None or len(services) < 1):
+        ansible_module.fail_json(msg="At least one name or services is "
+                                     "required")
     check_parameters(ansible_module, state, action, names)
 
+    # Use services if names is None
+    if services is not None:
```

Comment:
Comment doesn't match test. I'd prefer to have `if names is None:`.

Remember that one of names or services is not `None` at this point, as `require_one_of: ['name', 'serviecs']` is used.

Reply:
The parameters are even mutually exclusive: `mutually_exclusive=[["name", "services"]]`